xrootd/xrdproto/protocol.go

```diff
@@ -182,3 +183,18 @@ func (o *SecurityOverride) UnmarshalXrd(dec *xrdenc.RBuffer) error {
 	o.RequestLevel = RequestLevel(dec.ReadU8())
 	return nil
 }
+
+// SetOpaque appends opaque data to the provided path.
+func SetOpaque(path, opaque string) string {
```

Comment:
I think the name `SetOpaque` is confusing.
the doc string above would suggest a better name to be `AppendOpaque`.
but it's not quite that, as it replaces the opaque part in `path` with what is provided with `opaque`.
hum...

perhaps the confusing part is that `SetOpaque` returns the new value instead of adding it inplace?
it definitely looks confusing at call site:
```go
req.Path = xrdproto.SetOpaque(req.Path, opaque)
```

perhaps if it was:
```go
xrdproto.SetOpaque(&req.Path, opaque)
```
it would look better (and we could keep the name `SetOpaque`.)

what do you think?
ok to change to:
```go
func SetOpaque(path *string, opaque string) string { ... }
```
?

(alternatively, we could introduce a `type Opaque string` with methods attached to handle this...)